python/lsst/pipe/tasks/loadReferenceCatalog.py

```diff
@@ -265,12 +265,7 @@ def _formatCatalog(self, refCat, filterList):
         npRefCat['refMagErr'][:, :] = 99.0
 
         if self.config.doApplyColorTerms:
-            if isinstance(self.refObjLoader, ReferenceObjectLoader):
-                # Gen3
-                refCatName = self.refObjLoader.config.value.ref_dataset_name
-            else:
-                # Gen2
-                refCatName = self.refObjLoader.ref_dataset_name
+            refCatName = self.refObjLoader.config.ref_dataset_name
```

Comment:
Ah, this is a subtle one.  This is because the refobjloader config was changed from a ConfigurableField (retargeted task) to a ConfigField (just a regular subconfig).  And I think that either usage would have worked anyway.  The more frustrating thing, though, is that we weren't supposed to be using the `ref_dataset_name` anymore anyway, and I need to discuss with people about what to do about this.  (Thanks for the reminder!)

Reply:
The challenge here is that in gen3 the dataset name comes from the Task Connections. `LoadReferenceCatalogTask` doesn't have Connections; it's using the gen2 approach of specifying the refObjLoader as a config. A `ReferenceObjectLoader` doesn't actually know what the name of the refcat it's loading is; maybe we should fix that? We could add a `name=` kwarg to `init`, but I'm not sure how that would be filled in if `runQauntum` is not overridden (though I don't know if there are any such cases?).

Reply:
This is now captured in DM-35871.